hkmc2/shared/src/main/scala/hkmc2/codegen/js/JSBuilder.scala

```diff
@@ -466,7 +481,19 @@ class JSBuilder(using TL, State, Ctx) extends CodeBuilder:
         then "./" + os.Path(path).relativeTo(wd).toString
         else path
       doc"""import ${getVar(i._1)} from "${relPath}";"""
-    imps.mkDocument(doc" # ") :/: block(p.main, endSemi = false).stripBreaks :: (
+    val syms = internalSymbols match
```

Comment:
I'm against this overcomplicated mess. I think `program` should always create the bindings in the output. And the symbols should just be stored in `State`, just like `runtimeSymbol`. There's no reason to create new symbols every time and use mutable state like this.

Reply:
Yes it's fine to put them there.

Just separate the part to initialize the symbols form the rest of `program`'s logic, and use the appropriate thing in DiffTest. It's a tiny refactoring, and that's a million times better than adding some pointless mutable state.